data/txHandler.go

```diff
@@ -236,13 +254,24 @@ func (handler *TxHandler) backlogWorker() {
 			if !ok {
 				return
 			}
+			if wi.capguard != nil {
+				if err := wi.capguard.Release(); err != nil {
+					logging.Base().Warnf("Failed to release capacity to ElasticRateLimiter: %v", err)
+				}
+			}
 			if handler.checkAlreadyCommitted(wi) {
 				transactionMessagesAlreadyCommitted.Inc(nil)
+				if wi.capguard != nil {
+					wi.capguard.Served()
+				}
 				continue
 			}
 
 			// enqueue the task to the verification pool.
 			handler.txVerificationPool.EnqueueBacklog(handler.ctx, handler.asyncVerifySignature, wi, nil)
+			if wi.capguard != nil {
+				wi.capguard.Served()
+			}
```

Comment:
Several points here:
1. Served and Release both return err. One is checked, others are not. Why return an error if it is not important? 
2. When Release returns an error, it is of no consequence here. The logging of the warning can be inside Release. 
3. Why is it necessary to have 3 calls here? It is complicating the backlog code so much. There has to be another messaging mechanism.